content/technotes/2019-12-19-urls-checking/index.md

```diff
@@ -0,0 +1,456 @@
+---
+slug: "urls-tidying"
+title: "We cleaned our website URLs with R!"
+date: 2019-12-19
+authors:
+  - Maëlle Salmon
+  - Scott Chamberlain
+  - Stefanie Butland
+categories:
+  - technotes
+topicid: 1307
+tags:
+  - Markdown
+  - RMarkdown
+  - xml2
+  - commonmark
+  - crul
+  - hugo
+output: 
+  md_document:
+    variant: markdown_github
+    preserve_yaml: true
+---
+
+Last year we reported on [the joy of using commonmark and xml2 to parse
+Markdown content, like the source of this website built with
+Hugo](/technotes/2018/09/05/commonmark/), in particular [to extract
+links](https://ropensci.org/technotes/2018/09/05/commonmark/#urls-parsing),
+at the time merely to count them. How about we go a bit further and use
+the same approach to find links to be fixed? In this tech note we shall
+report our experience using R to find broken/suboptimal links and fix
+them.
+
+What is a bad URL?
+------------------
+
+We tackled a few URL issues:
+
+-   We had used absolute links (using our domain name) instead of
+    relative links which is in particular bad when looking at a preview
```

Comment:
I would omit this phrase because many people won't understand or need to understand " which is in particular bad when looking at a preview after a change"